harper-tree-sitter/src/lib.rs

```diff
@@ -115,29 +115,29 @@ impl Masker for TreeSitterMasker {
     }
 }
 
-/// Converts a set of byte-indexed [`Span`]s to char-index Spans, in-place.
+/// Converts a set of byte-indexed [`Span`]s to char-index Spans and returns them.
 /// NOTE: Will sort the given slice by their [`Span::start`].
 ///
-/// If any spans overlap, it will remove the second one.
-fn byte_spans_to_char_spans(byte_spans: &mut Vec<Span>, source: &str) {
-    byte_spans.sort_by_key(|s| s.start);
-
-    let cloned = byte_spans.clone();
-
-    let mut i: usize = 0;
-    byte_spans.retain(|cur| {
-        i += 1;
-        if let Some(prev) = cloned.get(i.wrapping_sub(2)) {
-            !cur.overlaps_with(*prev)
-        } else {
-            true
+/// If any spans overlap, it will merge them.
```

Comment:
This seemed more correct than ignoring the subsequent spans and didn't break any tests.

Original implementation was not working with the newer version of some grammars which have top-level items for comments and then sub-items for each line. Starting with 2 lines, the older logic was not sound (bacause 3 or more spans were overlapping) so I re-implemented it.